physics/discrete_trajectory_segment_body.hpp

```diff
@@ -107,9 +107,16 @@ void DiscreteTrajectorySegment<Frame>::clear() {
 template<typename Frame>
 typename DiscreteTrajectorySegment<Frame>::iterator
 DiscreteTrajectorySegment<Frame>::find(Instant const& t) const {
+  auto const it = FindImpl(t);
+  return it.has_value() ? *it : end();
+}
+
+template <typename Frame>
+std::optional<typename DiscreteTrajectorySegment<Frame>::iterator>
+DiscreteTrajectorySegment<Frame>::FindImpl(Instant const& t) const {
```

Comment:
We try to use the same order for the methods in the `.hpp` and `.cpp` files, so these implementation methods, which are private, should move down.

Reply:
Done.